lib/rules/at-rule-descriptor-value-no-unknown/index.mjs

```diff
@@ -36,6 +34,10 @@ const rule = (primary) => {
 
 				const { prop, value } = decl;
 
+				const lexer = result.stylelint.lexer;
+
+				if (!lexer) throw new Error('Expected a "lexer" object');
```

Comment:
Once we have new meta boolean field, we can check this earlier.
i.e. this check should not be done in the rules themselves

Whether the check is necessary or not is another discussion; I am just talking about the location.

Reply:
I added a getter, but I don't fully understand the significance of "early checks" in this context.

[Add getLexer](https://github.com/stylelint/stylelint/pull/8297/commits/41d9857043804ee7344adb6ea53620cee6a05951)

Reply:
I meant that we want to have this code once in the codebase.
i.e. if you have access to the enabled rules that have the new meta field you would run that function once or in an existing loop
```
// example
function foo(filteredRules, ruleMetadata) {
  // 1. loop on rules
  // 2. switch/case on lexers string ENUM (will only be an if since we only have one)
  // 3. throw
    // 3.1 wrong lexer
    // 3.2 missing lexer
}
```

Based on that knowledge the function should probably be called somewhere where you have these.
i.e. rules and meta
That file is probably lintPostcssResult.mjs.
e.g.
```js
/* […] */
// after line 89
checkLexers(ruleFunction.meta); // or checkLexer(ruleFunction.meta['bar']);
```
i.e. before you reach l110-123